extension/src/cli/git/reader.test.ts

```diff
@@ -54,6 +54,7 @@ describe('GitReader', () => {
       expect(mockedCreateProcess).toHaveBeenCalledWith({
         args: ['branch'],
         cwd,
+        env: { LANG: 'en_US.UTF-8' },
```

Comment:
[F] Rather than try to replace all possible languages' version of `HEAD detached at` it seems smarter to always ask for the English version and convert that.

Reply:
Makes sense! Though I wonder how other extensions handle things like this 🤔

Reply:
Gitlens:

<img width="515" alt="image" src="https://github.com/iterative/vscode-dvc/assets/37993418/010d6a7f-8e05-46ec-b93f-9c9deedc96be">

<img width="608" alt="image" src="https://github.com/iterative/vscode-dvc/assets/37993418/24801f10-7757-46d2-9009-44b77056b2f4">

<img width="595" alt="image" src="https://github.com/iterative/vscode-dvc/assets/37993418/eeae360d-c797-4528-bf16-6c94dcd6e755">

tl;dr - if the branch name has whitespace mangle the sha into `(${short sha}...)`

Reply:
We can overhaul when we get to #4111